src/libraries/Common/src/Interop/Windows/SspiCli/SecuritySafeHandles.cs

```diff
@@ -318,6 +317,59 @@ public new IntPtr DangerousGetHandle()
 
             return errorCode;
         }
+
+        public static unsafe int AcquireCredentialsHandle(
+            string package,
+            Interop.SspiCli.CredentialUse intent,
+            Interop.SspiCli.SCH_CREDENTIALS* authdata,
+            out SafeFreeCredentials outCredential)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(null, package, intent, (IntPtr)authdata);
+
+            int errorCode = -1;
+            long timeStamp;
+
+            // If there is a certificate, wrap it into an array.
+            // Not threadsafe.
+            IntPtr copiedPtr = authdata->paCred;
+            try
+            {
+                IntPtr certArrayPtr = new IntPtr(&copiedPtr);
+                if (copiedPtr != IntPtr.Zero)
+                {
+                    authdata->paCred = certArrayPtr;
+                }
+
+                outCredential = new SafeFreeCredential_SECURITY();
+
+                errorCode = Interop.SspiCli.AcquireCredentialsHandleW(
+                                null,
+                                package,
+                                (int)intent,
+                                null,
+                                authdata,
+                                null,
+                                null,
+                                ref outCredential._handle,
+                                out timeStamp);
+            }
+            finally
+            {
+                authdata->paCred = copiedPtr;
+            }
+
+#if TRACE_VERBOSE
```

Comment:
I thought we got rid of 'TRACE_VERBOSE'.  It's an ancient relic from .NET Framework and the 'TRAVE' environment variable ("TRAce + VErbose").

We should stop using it. Most developers don't build with that env variable set. Eventually we should transition all of the TRACE_VERBOSE stuff.

Instead, if we need more "chatty" diagnostic tracing, we should add a `NetEventSource.Verbose` method and map that to EventSource `verbose` level events.  The current `NetEventSource.Info` method maps to EventSource `info` level events.

Reply:
This is clone of the other overloads. Perhaps TRACE_VERBOSE should be removed completely in separate PR.

Reply:
fixed.